decoder/internal/schemahelper/block_schema.go

```diff
@@ -38,23 +38,24 @@ func MergeBlockBodySchemas(block *hcl.Block, blockSchema *schema.BlockSchema) (*
 		}
 		for bType, block := range depSchema.Blocks {
 			if _, exists := mergedSchema.Blocks[bType]; !exists {
+				b := block.Copy()
 				// propagate DynamicBlocks extension to any nested blocks
 				if mergedSchema.Extensions != nil && mergedSchema.Extensions.DynamicBlocks {
-					if block.Body.Extensions == nil {
-						block.Body.Extensions = &schema.BodyExtensions{}
+					if b.Body.Extensions == nil {
+						b.Body.Extensions = &schema.BodyExtensions{}
 					}
-					block.Body.Extensions.DynamicBlocks = true
+					b.Body.Extensions.DynamicBlocks = true
 				}
 
-				mergedSchema.Blocks[bType] = block
+				mergedSchema.Blocks[bType] = b
```

Comment:
Could we be a little more descriptive in variable names? I know it's against convention to have more than one character variables, but IMHO it improves readability here as `b` and `block` look so similar in meaning at a glance

Reply:
Renamed it to `copiedBlock` as per our offline discussion ✅